var/spack/repos/builtin/packages/cgm/package.py

```diff
@@ -25,30 +25,51 @@
 from spack import *
 
 
-class Cgm(Package):
+class Cgm(AutotoolsPackage):
     """The Common Geometry Module, Argonne (CGMA) is a code library
        which provides geometry functionality used for mesh generation and
        other applications."""
-    homepage = "http://trac.mcs.anl.gov/projects/ITAPS/wiki/CGM"
-    url      = "http://ftp.mcs.anl.gov/pub/fathom/cgm13.1.1.tar.gz"
+    homepage = "http://sigma.mcs.anl.gov/cgm-library"         
+    url = "http://ftp.mcs.anl.gov/pub/fathom/cgm-16.0.tar.gz"
 
+    version('16.0', 'a68aa5954d82502ff75d5eb91a29a01c')
     version('13.1.1', '4e8dbc4ba8f65767b29f985f7a23b01f')
     version('13.1.0', 'a6c7b22660f164ce893fb974f9cb2028')
     version('13.1', '95f724bda04919fc76818a5b7bc0b4ed')
 
-    depends_on("mpi")
+    variant("mpi", default=False, description='enable mpi support')
+    variant("oce", default=False, description='enable oce geometry kernel')
+
+    depends_on('mpi', when='+mpi')
+    depends_on('oce+X11', when='+oce')
+
+    def url_for_version(self, version):
+        if Version('13.0') <= version <= Version('13.999'):
+            return "http://ftp.mcs.anl.gov/pub/fathom/cgm{0}.tar.gz".format(version)
+        else:
+            return "http://ftp.mcs.anl.gov/pub/fathom/cgm-{0}.tar.gz".format(version)
 
     def patch(self):
-        filter_file('^(#include "CGMParallelConventions.h")',
-                    '//\1',
-                    'geom/parallel/CGMReadParallel.cpp')
-
-    def install(self, spec, prefix):
-        configure("--with-mpi",
-                  "--prefix=%s" % prefix,
-                  "CFLAGS=-static",
-                  "CXXFLAGS=-static",
-                  "FCFLAGS=-static")
-
-        make()
-        make("install")
+        if self.version < Version('14.1'):
+            filter_file('^(#include "CGMParallelConventions.h")',
+                        '//\1',
+                        'geom/parallel/CGMReadParallel.cpp')
+
+    def configure_args(self):
+        spec = self.spec
+
+        args = [
+        ]
+
+        if '+mpi' in spec:
+            args.extend([
+                "--with-mpi",
+                "CC={0}".format(spec['mpi'].mpicc),
+                "CXX={0}".format(spec['mpi'].mpicxx)
+            ])
+
+        if '+oce' in spec:
+            args.extend([
+                "--with-occ={0}".format(spec['oce'].prefix)
```

Comment:
If you're only adding a single element to the list, you can use `args.append()` instead of `args.extend([])`. Then you don't need to create an additional list.